src/NixTree/StorePath.hs

```diff
@@ -160,6 +167,7 @@ getPathInfo nixStore nixVersion options names = do
                 ++ ["--impure" | options & pioIsImpure]
                 ++ ["--recursive" | options & pioIsRecursive]
                 ++ ["--derivation" | (options & pioIsDerivation) && nixVersion >= Nix2_4]
+                ++ (if options & pioIsCustomStore then ["--store", unNixStore nixStore] else [])
```

Comment:
This is also another clue for me - if we think our `nixStore` is always correct, then why do we only pass it when it is "custom"? I feel like if we had the right abstraction we could always pass it.

Reply:
You are absolutely right! I thought the default is `--store /nix/store` (wrong) but actually the default is `--store auto`. See the updated code!